src/Raven.Server/Rachis/CandidateAmbassador.cs

```diff
@@ -111,12 +111,9 @@ private void Run()
                         Action disconnect;
                         try
                         {
-                            using (_engine.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
-                            {
-                                var connection = _engine.ConnectToPeer(_url, _certificate, context).Result;
-                                stream = connection.Stream;
-                                disconnect = connection.Disconnect;
-                            }
+                            var connection = _engine.ConnectToPeer(_url, _tag, _certificate).Result;
```

Comment:
I think it would be safer to use AsyncHelpers.RunSync

Reply:
why? This a dedicated thread for the candidate ambassador.